pdc_client/__init__.py

```diff
@@ -89,8 +94,63 @@ def read_config_file(server_alias):
     return result
 
 
+class MultiPageBeanBag(BeanBag):
+
+    def is_multi_page_response(self, response):
+        """
+        Check if response is a dict or not
+        """
+        if isinstance(response, dict):
+            if set([u'count', u'next', u'results', u'previous']).issubset(set(response.keys())):
+                return True
+        return False
+
+    def url2path_ext(self, url):
+        """
+        @param url
+        returns path, params
+        """
+        parsed = urlparse(url)
+        return parse_qs(parsed.query)
+
+    def call(self, path, *args, **kwargs):
+        """
+        Re-write the call function of BeanBag.
+        Let it default get multi pages data if there is no page parameter.
+        """
+        try:
+            if len(args) == 0:
+                verb, body = "GET", None
+                result = self.make_request(path, verb, kwargs, body)
+
+                if self.is_multi_page_response(result) and ("page" not in kwargs.keys()):
+                    count = result["count"]
+                    multi_result = result['results']
+                    while result['next']:
+                        next_page = self.url2path_ext(result['next'])
+                        kwargs["page"] = next_page["page"][0]
+                        result = self.make_request(path, verb, kwargs, body)
+                        multi_result.extend(result['results'])
```

Comment:
OK, after look into the yield codes in get_pages, I think this have a better user experience when user uses pdc list command with the formatted output. I will change code with yield and commit then.